cpp/tests/copying/reverse_tests.cpp

```diff
@@ -0,0 +1,186 @@
+
+  cudf::test::fixed_width_column_wrapper<T> expected(
+    expected_values.begin(), expected_values.end(), expected_valids.begin());
+
+  cudf::table_view input_table{{input}};
+  auto p_ret = cudf::reverse(input_table);
+
+  EXPECT_EQ(p_ret->num_columns(), 1);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(p_ret->view().column(0), expected);
+}
+
+TYPED_TEST(ReverseTypedTestFixture, ZeroSizeInput)
+{
+  using T = TypeParam;
+  static_assert(cudf::is_fixed_width<T>() == true, "this code assumes fixed-width types.");
+
+  cudf::test::fixed_width_column_wrapper<T, int32_t> input(thrust::make_counting_iterator(0),
+                                                           thrust::make_counting_iterator(0));
+
+  cudf::test::fixed_width_column_wrapper<T, int32_t> expected(thrust::make_counting_iterator(0),
+                                                              thrust::make_counting_iterator(0));
+
+  cudf::table_view input_table{{input}};
+  auto p_ret = cudf::reverse(input_table);
+
+  EXPECT_EQ(p_ret->num_columns(), 1);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(p_ret->view().column(0), expected);
+}
+
+class ReverseStringTestFixture : public cudf::test::BaseFixture {
+};
+
+TEST_F(ReverseStringTestFixture, ReverseNullable)
+{
+  constexpr cudf::size_type num_values{20};
+
+  // generating string column and valid inputs
+  std::vector<std::string> input_values(num_values);
+  std::vector<bool> input_valids(num_values);
+  for (size_t i{0}; i < num_values; i++) {
+    input_values[i] = "#" + std::to_string(i);
+    input_valids[i] = (i % 2) == 0 ? true : false;
+  }
+
+  // we then generate the expected values
+  std::vector<std::string> expected_values;
+  std::vector<bool> expected_valids;
+  for (int i = 19; i > -1; i--) {
+    expected_values.push_back(input_values[i]);
+    expected_valids.push_back(input_valids[i]);
+  }
+
+  // we then create the necessary tables
+  auto input = cudf::test::strings_column_wrapper(
+    input_values.begin(), input_values.end(), input_valids.begin());
+
+  auto expected = cudf::test::strings_column_wrapper(
+    expected_values.begin(), expected_values.end(), expected_valids.begin());
+
+  cudf::table_view input_table{{input}};
+  auto p_ret = cudf::reverse(input_table);
+
+  EXPECT_EQ(p_ret->num_columns(), 1);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(p_ret->view().column(0), expected);
+}
+
+TEST_F(ReverseStringTestFixture, ZeroSizeInput)
+{
+  std::vector<std::string> input_values{};
+  auto input = cudf::test::strings_column_wrapper(input_values.begin(), input_values.end());
+
+  auto count = cudf::test::fixed_width_column_wrapper<cudf::size_type>(
+    thrust::make_counting_iterator(0), thrust::make_counting_iterator(0));
+
+  auto expected = cudf::test::strings_column_wrapper(input_values.begin(), input_values.end());
+
+  cudf::table_view input_table{{input}};
+  auto p_ret = cudf::reverse(input_table);
+
+  EXPECT_EQ(p_ret->num_columns(), 1);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(p_ret->view().column(0), expected);
+}
```

Comment:
clang-format should fix this for you automatically. Do you not have clang-format set up to format on save?

```suggestion
}

```

Reply:
Not at the moment, I am just running the command provided in the cudf `CONTRIBUTING` file. But I will set it up.